src/main/java/io/aiven/kafka/auth/json/AclPermissionType.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2023 Aiven Oy https://aiven.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.auth.json;
+
+public enum AclPermissionType {
+    ALLOW((byte) 1),
+    DENY((byte) 0);
+
+    private final byte code;
+
+    AclPermissionType(final byte code) {
+        this.code = code;
+    }
+
+    public org.apache.kafka.common.acl.AclPermissionType toNativeType() {
```

Comment:
Can we use the native types directly, do we need this wrapper? We may need some additional check to fail on `UNKNOWN` and `ANY`, but that's maybe a couple of lines

Reply:
I wrote about this here: https://github.com/aiven/auth-for-apache-kafka/pull/135#issuecomment-1629004156

> However I still left the `AclPermissionType` class because I did not wanted to couple too much with the Kafka Acl permission type, as they have also other values (ANY and UNKNOWN) which are not relevant to our use case.

But if you prefer to use `org.apache.kafka.common.acl.AclPermissionType` directly i'm also fine in changing it.

Reply:
Let's keep it, nvm